Improved Icestorm (Whispering Ice Unique skill) DPS

mbellifa opened the pull request:
I've been using this code as an ad hoc modification to PoB for a few months now and thought that others may benefit as well. It adds a skill part drop down for 1 Shard/All Shards, 1 Shard is essentially unchanged from how it calculated before with the exception of it now showing DPS since as far as I could tell I had to enable that for the whole skill. The second "All Shards" mode adds a DPS multiplier based on skill effect duration, this is to try to account for the ability for Icestorm to stack on the same area meaning that over time having more skill effect duration increases the dps by having multiple storms over the same point.

Let me know if there's a better way to do what I've done or if I need to format things differently either in the code or in future pull requests.

LocalIdentity commented:
When changing lines for skills in PoB, you need to change the lines by editing the export script other.txt instead of the actual lua file. Please update the pull with this in mind

mbellifa commented:
Aha. Sorry I didn't know that. I've added an updated other.txt from the Export directory but I'm having a heck of a time trying to figure out how to test it to make sure it updated the lua in the same way. Do I need to load the Export.sln into Visual Studio in order to verify that the resulting lua matches?

ppoelzl commented:
No VS needed, just use the exporter like described here: https://github.com/LocalIdentity/PathOfBuilding/blob/master/CONTRIBUTING.md#exporting-data-from-ggpk

mbellifa commented:
Thanks! I somehow misread that as pointing to the same Launch.lua as the general dev mode instructions. I've added one more commit to show DPS instead of average hit and it appears to be generating what I intended.

LocalIdentity commented on Data/3_0/Skills/other.lua:
```diff
@@ -1027,7 +1043,6 @@ skills["IcestormUniqueStaff12"] = {
 		"firestorm_drop_ground_ice_duration_ms",
 		"skill_art_variation",
 		"skill_effect_duration_per_100_int",
-		"base_skill_show_average_damage_instead_of_dps",
```
You can't remove this line from the other.lua file, everything in here must be modified by only changing the export script.
To get it to show DPS instead of average damage, just add this
`statMap = {
		["base_skill_show_average_damage_instead_of_dps"] = {
		},
	},`
above the base mods section

LocalIdentity replied:
You can look at the Storm Brand gem which does the same thing

mbellifa replied:
Thanks I reverted that commit and added hte line you suggested to the other.txt file.

mbellifa commented:
Sorry for the delay and how messy the commits have gotten. I updated my local dev branch to the upstream and reverted my problematic commits. Let me know if I'm missing anything else.

yuzi-co commented:
There is now a limit to 5 storms. This should be either updated or perhaps is redundant, since you can just multiply the average hit by 5.

ppoelzl commented:
Outdated as of game version 3.12.0.
